sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -567,6 +567,29 @@ public Duration getAllowedTimestampSkew() {
   @Target(ElementType.METHOD)
   public @interface ProcessElement {}
 
+  /**
+   * <b><i>Experimental - no backwards compatibility guarantees. The exact name or usage of this
+   * feature may change.</i></b>
+   *
+   * <p>Annotation that may be added to a {@link ProcessElement} or {@link OnTimer} method to
```

Comment:
Can you elaborate on what this means for a timer? At first glance, it seems like a timer is just "this timer you set fired", so it should never be unstable. Possible thoughts:

- Are the values of side-inputs observed in a stable way? If not, that should be clearly documented (maybe even enforced -- any DoFn that requires stable input should not be allowed to have side inputs).
- Are the set of elements that may have been observed (and thus put into state) stable?
- Something else I may be missing?

Reply:
The part relevant for a timer is that "the observable contents of... mutable state must be stable upon retries".